model/websocket_client.go

```diff
@@ -77,8 +89,12 @@ func NewWebSocketClient4(url, authToken string) (*WebSocketClient, *AppError) {
 }
 
 func (wsc *WebSocketClient) Connect() *AppError {
+	return wsc.ConnectWithDialer(websocket.DefaultDialer)
+}
+
+func (wsc *WebSocketClient) ConnectWithDialer(dialer *websocket.Dialer) *AppError {
```

Comment:
@jwilander, from what I can see, `Connect` doesn't get used within mattermost-server. I was going to ask if it made sense to be able to invoke `NewWebSocketClient4WithDialer` with one dialer and `Connect` with another, but it seems like this might be a "legacy constructor" of sorts.

Reply:
Yeah, I think this is fine. The general use case behind `Connect()` was to allow easy reconnecting of a disconnected client without having to make a new one. I don't why you'd connect a second time with a new dialer but I can't see any reason to not allow it